src/chain.h

```diff
@@ -337,6 +339,11 @@ class CBlockIndex
         return ((nStatus & BLOCK_VALID_MASK) >= nUpTo);
     }
 
+    //! Check whether the block associated with this index entry is pruned or not.
+    inline bool IsPruned() const {
```

Comment:
Adding this to the `chain` module introduces a cyclic dependency between `chain` and `validation` (as fHavePruned is defined there). There's no need for this to be a `CBlockIndex` member as it only uses public fields, so I would suggest making it a function in `validation` instead.

Reply:
Good point. I moved it (back) to `validation.h`.